src/sentry/models/groupassignee.py

```diff
@@ -69,11 +69,11 @@ def assign(
             ).update(**{assignee_type_attr: assigned_to_id, other_type: None, "date_added": now})
         else:
             affected = True
+
+        if affected:
             issue_assigned.send_robust(
```

Comment:
The `send_robust` function now runs and fires the issue assigned webhook if the assignee was affected instead of just when the it was created (ie. if the assignee was changed from one user to another AND if it was changed from unassigned to assigned)